src/gtc/oir.py

```diff
@@ -145,19 +145,68 @@ class Interval(LocNode):
     start: AxisBound
     end: AxisBound
 
+    @root_validator
+    def check(cls, values: Dict[str, Any]) -> Dict[str, Any]:
+        start, end = values["start"], values["end"]
+        if start.level == common.LevelMarker.END and end.level == common.LevelMarker.START:
+            raise ValueError("Start level must be smaller or equal end level")
+        if start.level == end.level and start.offset > end.offset:
+            raise ValueError("Start offset must be smaller or equal end offset")
+        return values
 
-class VerticalLoop(LocNode):
+
+class CacheDecl(LocNode):
```

Comment:
Maybe 
```suggestion
class CacheDesc(LocNode):
```
to distinguish it from the other `Decl`s which are semantically different.